jerry-core/ecma/operations/ecma-jobqueue.c

```diff
@@ -28,26 +34,221 @@
+  ecma_value_t capability = ecma_op_object_get (reaction_p, str_0);
+  /* 3. string '1' indicates the [[Handler]] of reaction */
+  ecma_value_t handler = ecma_op_object_get (reaction_p, str_1);
+
+  JERRY_ASSERT (ecma_is_value_boolean (handler) || ecma_op_is_callable (handler));
+
+  ecma_value_t handler_result;
+
+  if (ecma_is_value_boolean (handler))
+  {
+    /* 4-5. True indicates "identity" and false indicates "thrower" */
+    handler_result = ecma_copy_value (job_p->argument);
+  }
+
+  else
+  {
+    /* 6. */
+    handler_result = ecma_op_function_call (ecma_get_object_from_value (handler),
+                                            ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED),
+                                            &(job_p->argument),
+                                            1);
+  }
+
+  ecma_value_t status;
+  /* 7. */
+  if (ecma_is_value_false (handler) || ECMA_IS_VALUE_ERROR (handler_result))
+  {
+    /* str '2' indicates [[Reject]] of Capability */
+    handler_result = handler_result & ~ECMA_VALUE_ERROR_FLAG;
+    ecma_value_t reject = ecma_op_object_get (ecma_get_object_from_value (capability), str_2);
+
+    JERRY_ASSERT (ecma_op_is_callable (reject));
+
+    status = ecma_op_function_call (ecma_get_object_from_value (reject),
+                                    ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED),
+                                    &handler_result,
+                                    1);
+    ecma_free_value (reject);
+  }
+  /* 8 */
+  else
+  {
+    /* str '1' indicates [[Resolve]] of Capability */
+    ecma_value_t resolve = ecma_op_object_get (ecma_get_object_from_value (capability), str_1);
+
+    JERRY_ASSERT (ecma_op_is_callable (resolve));
+
+    status = ecma_op_function_call (ecma_get_object_from_value (resolve),
+                                    ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED),
+                                    &handler_result,
+                                    1);
+    ecma_free_value (resolve);
+  }
+
+  ecma_free_value (handler_result);
+  ecma_free_value (handler);
+  ecma_free_value (capability);
+  ecma_deref_ecma_string (str_0);
+  ecma_deref_ecma_string (str_1);
+  ecma_deref_ecma_string (str_2);
+  free_promise_reaction_job (job_p);
+
+  return status;
+} /* process_promise_reaction_job */
+
+/**
+ * Process the PromiseResolveThenableJob
+ *
+ * See also: ES2015 25.4.2.2
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value
+ */
+static ecma_value_t
+process_promise_resolve_thenable_job (void *obj_p) /**< the job to be operated */
+{
+  ecma_job_promise_resolve_thenable_t *job_p = (ecma_job_promise_resolve_thenable_t *) obj_p;
+  ecma_object_t *promise_p = ecma_get_object_from_value (job_p->promise);
+  ecma_promise_resolving_functions_t *funcs;
+  funcs = ecma_promise_create_resolving_functions (promise_p);
+
+  ecma_value_t argv[] = {funcs->resolve, funcs->reject};
```

Comment:
Sorry, what is the suitable style for the array? I didn't find examples in #1459